src/extractor/extractor.cpp

```diff
@@ -118,7 +118,7 @@ int Extractor::run(ScriptingEnvironment &scripting_environment)
     TIMER_START(extracting);
 
     const unsigned recommended_num_threads = tbb::task_scheduler_init::default_num_threads();
-    const auto number_of_threads = std::min(recommended_num_threads, config.requested_num_threads);
+    const auto number_of_threads = std::min(recommended_num_threads, std::max(config.requested_num_threads, 1u));
```

Comment:
Nope, I didn't.

One more thing I noticed was that `Extractor::run` initialises the `tbb::task_scheduler_init` based on the config, while `Contractor::Run` does not (seems like leaving it as user's responsibility). I'm unsure, but looks like overlooked interface unification or perhaps by-design. I know too little of the OSRM, didn't want to draw too far-reaching conclusions :)